packages/cli/src/InternalHooks.ts

```diff
@@ -300,47 +314,41 @@ export class InternalHooks implements IInternalHooksClass {
 
 		const promises = [];
 
-		const properties: IExecutionTrackProperties = {
+		const telemetryProperties: IExecutionTrackProperties = {
 			workflow_id: workflow.id,
 			is_manual: false,
 			version_cli: N8N_VERSION,
 			success: false,
 		};
 
 		if (userId) {
-			properties.user_id = userId;
+			telemetryProperties.user_id = userId;
 		}
 
 		if (runData?.data.resultData.error?.message?.includes('canceled')) {
 			runData.status = 'canceled';
 		}
 
-		properties.success = !!runData?.finished;
+		telemetryProperties.success = !!runData?.finished;
```

Comment:
Since we added `status` to `IRun` can't we just check whether `runData.status === 'success'`?

Reply:
Not the way it is set up right now, since internal and external hooks run in parallel and what is being updated with the final status is not the rundata object, but the database row only.
I'll look into whether we can sort this correctly so that the data reaching internalhooks also contains the final status